### PR TITLE
TandemFoil Paper: lr=1e-4 at champion config

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Lower LR (1e-4 vs 1.25e-4) may help TFP converge to a better basin while reducing the instability at cosine LR peaks. haku #3056 is sweeping LR around the champion — this is a complementary run at lr=1e-4.

## Instructions

```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil_paper --optimizer lion --lr 1e-4 --cosine-t-max 10 \
  --grad-clip 0.5 --weight-decay 1e-2 --enable-fourier \
  --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --ema-decay 0.999 \
  --wandb_group tfp-lr-sweep
```

## Reporting Requirements
- Best val `field_mse` + epoch
- Test `field_mse` at best val checkpoint
- W&B run ID

## Baseline

**TFP baselines:**
- val best: `field_mse = 0.002383` @ ep443 (PR #3025, W&B: `d1xh0o1p`)
- Config: Lion lr=1.25e-4, T_max=10, gc=0.5, EMA=0.999, 3L/192d, Fourier+physics
- paper target: normalized full-field MSE (field_mse, lower is better)
- **No clean test_primary/field_mse yet** — that is what the TFP experiments must establish

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT shinobu:

## Results

### Experiment: TFP lr=1e-4 (champion config)

**Hypothesis**: Lower LR (1e-4 vs baseline 1.25e-4) may help TFP converge to a better basin while reducing instability at cosine LR peaks.

**Result: NEGATIVE** — `field_mse` never resolved to a finite value across 393 epochs (training still running, reporting early as result is conclusive).

| Metric | Baseline (lr=1.25e-4) | This run (lr=1e-4) |
|---|---|---|
| **val_primary/field_mse** | **0.002383** (ep443) | **Infinity** (all 393 epochs) |
| val/surface_mse_Ux | — | 0.0012 (healthy) |
| val/surface_mse_Uy | — | 0.0033 (healthy) |
| val/surface_mse_p | — | Infinity (periodic finite troughs, best ~5e12 at ep379) |
| val/volume_mse_Ux | — | 0.0041 (healthy) |
| val/volume_mse_Uy | — | 0.0033 (healthy) |
| val/volume_mse_p | — | **Infinity (NEVER finite)** |
| Training loss | — | 0.033 (converged well) |
| Peak memory | — | ~3.4 GB |

### Command
```bash
cd /workspace/senpai/target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset tandemfoil_paper --optimizer lion --lr 1e-4 --cosine-t-max 10 \
  --grad-clip 0.5 --weight-decay 1e-2 --enable-fourier \
  --model-layers 3 --model-hidden-dim 192 --model-heads 3 \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --epochs 999 --ema-decay 0.999 \
  --agent shinobu --wandb-name "shinobu/tfp-lr-1e4" --wandb-group "tfp-lr-sweep"
```

**W&B run ID**: `z63sarn6`

### What happened

The model trained well in normalized space — training loss converged to 0.033, velocity channels (Ux, Uy) achieved excellent metrics on both surface and volume. The problem is exclusively in the **pressure channel** during evaluation denormalization.

With `asinh_pressure` enabled, the eval path applies `torch.sinh()` to denormalize pressure predictions. When the model's normalized pressure outputs have extreme outlier values, `sinh()` overflows to Infinity. This is a known early-training artifact — the baseline at lr=1.25e-4 resolved this by epoch 443 as the model constrained its outputs.

**At lr=1e-4, the pressure never converged enough to exit the overflow regime**:
- **Surface pressure MSE**: oscillated wildly (periodic troughs down to ~5e12, but mostly Infinity), showing the cosine LR cycling causes some epochs to briefly avoid overflow
- **Volume pressure MSE**: was **never once finite** in 393 epochs — the volume field predictions remain too extreme for sinh() throughout

The 20% lower learning rate appears to slow the pressure channel convergence rate below the critical threshold needed to escape the sinh overflow regime within a reasonable training budget. The baseline at lr=1.25e-4 needed 443 epochs to resolve this; extrapolating from the surface pressure trough trajectory, lr=1e-4 might need 600+ epochs — well beyond our budget.

### Suggested follow-ups
- **lr=1.15e-4**: If the goal is to probe below 1.25e-4, try a smaller reduction. The convergence-to-overflow boundary may be narrow.
- **Pressure-specific gradient scaling**: Rather than lowering global LR, selectively reduce pressure channel learning dynamics.
- **Clamp before sinh**: Adding `torch.clamp(pred, -20, 20)` before `sinh()` would prevent overflow while still allowing the model to converge. This is a code-level fix rather than a hyperparameter change.
- **Longer warmup**: lr=1e-4 might work with a linear warmup phase to bootstrap pressure convergence before cosine cycling begins.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3096 because ADVISOR: Closing — lr=1e-4 is a dead end on TFP. val_primary/field_mse was Infinity for all 393 epochs — the pressure channel NEVER became finite (0 finite values out of 393). At 20% lower LR than champion (1.25e-4), the model converges too slowly for the asinh-transformed pressure targets to resolve within the training budget. Velocity channels converge normally (Ux=0.00128, Uy healthy), confirming this is a pressure convergence speed issue. LR=1.25e-4 is confirmed as near the minimum viable LR for TFP. The LR improvement trend (3e-4→1.25e-4) stops here. Good diagnosis, shinobu..
